File: testbed/test/cruise/test/UnidirectionalMStarTest.java
package cruise.test;

import org.junit.*;

public class UnidirectionalMStarTest
{

  @Test(expected=RuntimeException.class)
  public void constructorTooFew()
  {
    StudentAL s = new StudentAL(99);
    StudentAL s2 = new StudentAL(98);
    new MentorAL("blah",s,s2);
  }
  
  
  @Test
  public void constructorRequiresMinimum()
  {
    StudentAL s = new StudentAL(99);
    StudentAL s2 = new StudentAL(98);
    StudentAL s3 = new StudentAL(97);
    
    MentorAL m = new MentorAL("blah",s,s2,s3);
    Assert.assertEquals(3,m.numberOfStudents());
  }
  
  @Test
  public void addRemoveWithinLimits()
  {
    StudentAL s = new StudentAL(99);
    StudentAL s2 = new StudentAL(98);
    StudentAL s3 = new StudentAL(97);
    StudentAL s4 = new StudentAL(96);
    StudentAL s5 = new StudentAL(95);
    StudentAL s6 = new StudentAL(94);
    
    MentorAL m = new MentorAL("blah",s,s2,s3);
    Assert.assertEquals(3,m.numberOfStudents());
    Assert.assertEquals(true,m.addStudent(s4));
    Assert.assertEquals(true,m.addStudent(s5));
    
    Assert.assertEquals(5,m.numberOfStudents());
    
    Assert.assertEquals(false,m.removeStudent(s6));
    Assert.assertEquals(true,m.removeStudent(s3));
    Assert.assertEquals(true,m.removeStudent(s4));
    Assert.assertEquals(false,m.removeStudent(s));
    Assert.assertEquals(3,m.numberOfStudents());
  }
  
  
  @Test
  public void deleteDoesNotChangeStudent()
  {
    StudentAL s = new StudentAL(99);
    StudentAL s2 = new StudentAL(98);
    StudentAL s3 = new StudentAL(98);

    ProgramAL p = new ProgramAL();
    s.setProgram(p);
    
    MentorAL m = new MentorAL("blah",s,s2,s3);
    m.delete();
    Assert.assertEquals(0,m.numberOfStudents());
    Assert.assertEquals(p,s.getProgram());
  }
}

File: testbed_python/test/associations/unidirectional_m_star_test.py
import unittest

# Test classes are imported dynamically
# Must be generated into ImportModules' namespace then imported
from ImportModules import importModules

importModules(["MentorAL", "ProgramAL", "StudentAL"], ["associations"])
from ImportModules import *

class UnidirectionalMStarTest(unittest.TestCase):
    def test_constructorTooFew(self):
        s = StudentAL.StudentAL(99)
        s2 = StudentAL.StudentAL(98)
        with self.assertRaises(RuntimeError):
            MentorAL.MentorAL("blah",s,s2)
    
    def test_constructorRequiresMinimum(self):
        s = StudentAL.StudentAL(99)
        s2 = StudentAL.StudentAL(98)
        s3 = StudentAL.StudentAL(97)
        
        m = MentorAL.MentorAL("blah",s,s2,s3)
        self.assertEqual(3,m.numberOfStudents())
    
    def test_addRemoveWithinLimits(self):
        s = StudentAL.StudentAL(99)
        s2 = StudentAL.StudentAL(98)
        s3 = StudentAL.StudentAL(97)
        s4 = StudentAL.StudentAL(96)
        s5 = StudentAL.StudentAL(95)
        s6 = StudentAL.StudentAL(94)
        
        m = MentorAL.MentorAL("blah",s,s2,s3)
        self.assertEqual(3,m.numberOfStudents())
        self.assertEqual(True,m.addStudent(s4))
        self.assertEqual(True,m.addStudent(s5))
        
        self.assertEqual(5,m.numberOfStudents())
        
        self.assertEqual(False,m.removeStudent(s6))
        self.assertEqual(True,m.removeStudent(s3))
        self.assertEqual(True,m.removeStudent(s4))
        self.assertEqual(False,m.removeStudent(s))
        self.assertEqual(3,m.numberOfStudents())

    def test_deleteDoesNotChangeStudent(self):
        s = StudentAL.StudentAL(99)
        s2 = StudentAL.StudentAL(98)
        s3 = StudentAL.StudentAL(98)

        p = ProgramAL.ProgramAL()
        s.setProgram(p)
        
        m = MentorAL.MentorAL("blah",s,s2,s3)
        del m
        self.assertEqual(p,s.getProgram())
